src/axes/timeAxis.ts

```diff
@@ -392,9 +392,14 @@ export module Axes {
       let tickLabelsEnter = tickLabels.enter().append("g").classed(Axis.TICK_LABEL_CLASS, true);
       tickLabelsEnter.append("text");
       let xTranslate = (this._tierLabelPositions[index] === "center" || config.step === 1) ? 0 : this.tickLabelPadding();
-      let yTranslate = this.orientation() === "bottom" ?
-          d3.sum(this._tierHeights.slice(0, index + 1)) - this.tickLabelPadding() :
-          this.height() - d3.sum(this._tierHeights.slice(0, index)) - this.tickLabelPadding();
+      let yTranslate: number;
+      if (this.orientation() === "bottom") {
+        yTranslate = d3.sum(this._tierHeights.slice(0, index + 1)) - this.tickLabelPadding();
+      } else if (this._tierLabelPositions[index] === "center") {
+        yTranslate = this.height() - d3.sum(this._tierHeights.slice(0, index)) - this.tickLabelPadding() - this._maxLabelTickLength();
+      } else {
+        yTranslate = this.height() - d3.sum(this._tierHeights.slice(0, index)) - this.tickLabelPadding();
+      }
```

Comment:
Just curious, it does not seem that the center/top case is being considered.  If it is, it doesn't read to be so.  Can we have 2 `if` blocks?  One for standard orientation bottom vs top and then another to account for the center?